### PR TITLE
Solve bug

### PR DESCRIPTION
**Pull request by rorpyll**

### CODE REVIEW

File: app/controllers/attendences_controller.rb [hunk 1]
```diff
@@ -4,18 +4,10 @@
...
-      end
+    if current_employee.role.name == 'HR'
+        @attendences = Attendence.order(created_at: :desc)
     else
-      unless params[:start_date].blank? && params[:end_date].blank?
-        @attendences = Attendence.where(employee_id: current_employee, created_at: params[:search]..params[:sear])
-      else
-        @attendences = Attendence.all
-      end
+        @attendences = Attendence.where(employee_id: current_employee).order(created_at: :desc)
```
Comment (salvimohan) [inline app/controllers/attendences_controller.rb:10]: current_employees.attendences we can use

File: app/controllers/attendences_controller.rb [hunk 2]
```diff
@@ -4,18 +4,10 @@
 class AttendencesController < InheritedResources::Base
   #load_and_authorize_resource
   def index
-    if current_user.role.name == 'HR'
-      unless params[:start_date].blank? && params[:end_date].blank?
-        @attendences = Attendence.where(created_at: params[:start_date]..params[:end_date])
-      else
-        @attendences = Attendence.all 
-      end
+    if current_employee.role.name == 'HR'
```
Comment (salvimohan) [inline app/controllers/attendences_controller.rb:7]: current_employee&.role&.name

File: app/controllers/attendences_controller.rb [hunk 3]
```diff
@@ -29,25 +21,49 @@ def edit
...
-    last = Attendence.where(checkin_time: Time.zone.now-2.minutes..Time.zone.now).last
-    unless last.nil?
-       last.update(checkout_time: nil)
+    last_attendance = Attendence.where(checkin_time: Time.zone.now-2.minutes..Time.zone.now).last
+    unless last_attendance.nil?
+      last_attendance.update_column('checkout_time', nil)
     else
       @attendence = Attendence.create(employee_id: current_employee.id, checkin_time: Time.zone.now, status: 'Present')
-      @attendence.save
+      @attendence.save(validate: false)
```
Comment (salvimohan) [inline app/controllers/attendences_controller.rb:29]: if you are using create than why @attendence.save(validate: false)

File: app/controllers/daily_tasks_controller.rb [hunk 4]
```diff
@@ -5,11 +5,16 @@ def index
 	end
 
 	def show 
-		@daily_task = DailyTask.find(params[:id])
+			@daily_task = DailyTask.find(params[:id])
```
Comment (salvimohan) [inline app/controllers/daily_tasks_controller.rb:8]: rubocop?

File: app/controllers/tickets_controller.rb [hunk 5]
```diff
@@ -19,26 +19,34 @@ def create
 
   def update
```
Comment (salvimohan) [inline app/controllers/tickets_controller.rb:20]: why you are using the two action for updating status

File: app/models/attendence.rb [hunk 6]
```diff
@@ -16,16 +17,10 @@ def working_hour
     end
   end
 
+  def checkout_time_validation
+    return errors.add :base, "Checkout time can't be blank"  unless checkout_time.present?
```
Comment (salvimohan) [inline app/models/attendence.rb:21]: rubocop

File: app/views/attendences/search.js.erb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+
+$(".xyz").html("<%= j render partial: "search" %>");
```
Comment (salvimohan) [inline app/views/attendences/search.js.erb:2]: why xyz?

File: app/controllers/tickets_controller.rb (hunk 5 above)
Comment (rorpyll) [reply app/controllers/tickets_controller.rb:20]: both are changing status ,one is accept status and second is decline status

**Pull request merged**